src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
+    samples.
 
-def write_to_filestore(job, work_dir, ids, *filenames):
+    :param JobFunctionWrappingJob job: Toil Job instance
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to BAM file
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: None
     """
-    Given one or more file names in working directory, writes
-    files to filestore and stores the filestore promise in a dictionary
+    get_shared_files = job.wrapJobFn(download_shared_files, config).encapsulate()
+    samples = [(uuid, url, rg_line)]
+    run_pipeline = Job.wrapJobFn(run_gatk_germline_pipeline, samples, get_shared_files.rv())
+    get_shared_files.addChild(run_pipeline)
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are keys for ids
 
-    :returns: shared ids dictionary
+def run_gatk_germline_pipeline(job, samples, config):
     """
-    for filename in filenames:
-        ids[filename] = job.fileStore.writeGlobalFile(os.path.join(work_dir, filename))
-    return ids
-
-
-def read_from_filestore_hc(job, work_dir, ids, *filenames):
-    """
-    Reads file from fileStore and writes it to working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are filenames
-    """
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-
+    Configures and runs the GATK germline pipeline with filtering.
 
-def move_to_output_dir(work_dir, output_dir, *filenames):
-    """`
-    Moves files from the working directory to the output directory.
+    VQSR is done if the run_vqsr parameter is set to True or there are at least
+    30 samples. Otherwise, each sample is hard filtered using recommended GATK parameters.
 
-    :param work_dir: the working directory
-    :param output_dir: the output directory
-    :param filenames: remaining arguments are filenames
+    :param list samples: List of tuples (uuid, url, rg_line) for joint variant calling.
+                         (i.e. samples are grouped together for VQSR). Otherwise,
+                         samples are filtered using GATK recommended hard filters.
+    :param Namespace config: Input parameters and reference files
+    :param str suffix: Adds suffix to end of filename.
+    :return: None
     """
-    for filename in filenames:
-        origin = os.path.join(work_dir, filename)
-        dest = os.path.join(output_dir, filename)
-        shutil.move(origin, dest)
+    if len(samples) == 0:
+        raise ValueError('No samples were provided!')
 
+    cores = multiprocessing.cpu_count()
```

Comment:
CC @jvivian but I thought we decided against autodetecting core count moving forward. What was the end decision?